plugins/ccgx/fu-ccgx-i2c.c

```diff
@@ -0,0 +1,774 @@
+			     "failed to reset i2c:%s", 
+			     error_local->message);
+		return FALSE;
+	}
+	return TRUE;
+}
+
+static gboolean
+i2c_get_status (FuDevice *device, CyI2CDeviceHandle* handle, guint8 mode, guint8 *i2c_status, GError **error)
+{
+	g_autoptr(GError) error_local =	NULL;
+	GUsbDevice *usb_device = NULL;
+	guint32 io_timeout = FU_CCGX_I2C_WAIT_TIMEOUT;
+	guint16 scb_index = 0;
+	guint16 w_value, w_index, w_length;
+	guint8 bm_request;
+	gsize actual_length = 0;
+
+	g_return_val_if_fail (handle != NULL, FALSE);
+	g_return_val_if_fail (FU_USB_DEVICE (device), FALSE);
+	usb_device = fu_usb_device_get_dev (FU_USB_DEVICE (device));
+	g_return_val_if_fail (usb_device != NULL, FALSE);
+
+	scb_index = handle->inf_num;
+
+	if (scb_index > 0)
+		scb_index = 1;
+
+	bm_request = CY_I2C_GET_STATUS_CMD;
+	w_value = ((scb_index << CY_SCB_INDEX_POS) | mode);
+	w_index = 0;
+	w_length = CY_I2C_GET_STATUS_LEN;
+	if (!g_usb_device_control_transfer (usb_device,
+					   G_USB_DEVICE_DIRECTION_DEVICE_TO_HOST,
+					   G_USB_DEVICE_REQUEST_TYPE_VENDOR,
+					   G_USB_DEVICE_RECIPIENT_DEVICE,
+					   bm_request,
+					   w_value,
+					   w_index,
+					   (guint8*)&i2c_status,
+					   w_length,
+					   &actual_length,
+					   io_timeout,
+					   NULL,
+					   &error_local)) {
+		g_set_error (error,
+			     FWUPD_ERROR,
+			     FWUPD_ERROR_NOT_SUPPORTED,
+			     "failed to get i2c status:%s", 
+			     error_local->message);
+
+		return FALSE;
+	}
+	return TRUE;
+
+}
+
+static gboolean
+wait_for_notification (FuDevice *device, CyI2CDeviceHandle* handle, guint16 *bytes_pending, guint32 io_timeout, GError **error)
+{
+	g_autoptr(GError) error_local = NULL;
+	GUsbDevice *usb_device = NULL;
+	gsize actual_length = 0;
+	guint8 ep_num;
+	guint8 i2c_status[CY_I2C_EVENT_NOTIFICATION_LEN] = {0};
+
+	g_return_val_if_fail (handle != NULL, FALSE);
+	g_return_val_if_fail (FU_USB_DEVICE (device), FALSE);
+	usb_device = fu_usb_device_get_dev (FU_USB_DEVICE (device));
+	g_return_val_if_fail (usb_device != NULL, FALSE);
+
+	ep_num = handle->ep.intr_in;
+
+	if (g_usb_device_interrupt_transfer (usb_device,
+					     ep_num,
+					     (guint8*)i2c_status,
+					     CY_I2C_EVENT_NOTIFICATION_LEN,
+					     (gsize*)&actual_length,
+					     io_timeout,
+					     NULL,
+					     &error_local)) {
+
+		if (i2c_status[0] & CY_I2C_ERROR_BIT) {
+			if (i2c_status[0] & 0x80) { /* write */
+				if (i2c_reset (device,handle,CY_I2C_MODE_WRITE,NULL)  == FALSE) {
+					g_warning ("failed to reset i2c for write while getting i2c event");
+				}
+				memcpy(bytes_pending, &i2c_status[1], 2);
+
+			} else { /* read */
+				if (i2c_reset (device,handle,CY_I2C_MODE_READ,NULL) == FALSE) {
+					g_warning ("failed to reset i2c for read while getting i2c event");
+				}
+				memcpy(bytes_pending, &i2c_status[1], 2);
+			}
+			g_set_error (error,
+			     FWUPD_ERROR,
+			     FWUPD_ERROR_NOT_SUPPORTED,
+			     "i2c status error in i2c event : 0x%x", 
+			     (guint8) i2c_status[0]);
+
```

Comment:
extraneous whitepsace

Reply:
Do you mean "whitespace"? then please point out whitespace. e.g. (guint8) i2c_status[0] -> (guint8)i2c_status[0]

Reply:
line up the text like the style of rest of project

Reply:
I fixed it.